src/components/Admin/UserPage/DeleteOldDataModal.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  CheckboxGroup,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  useDisclosure,
+  Text,
+  Divider,
+  Flex,
+} from '@chakra-ui/react';
+import React, {useState} from 'react';
+import {useTranslation} from 'react-i18next';
+import {FaTrash} from 'react-icons/fa';
+import {useAuth} from '../../../contexts/AuthContext';
+import {firestore} from '../../../firebase';
+
+function DeleteOldDataModal(): JSX.Element {
+  // --------------- State maintenance variables ------------------------
+  const {isOpen, onOpen, onClose} = useDisclosure();
+  const [areDisclosuresChecked, setAreDisclosuresChecked] = useState([
+    false,
+    false,
+    false,
+  ]);
+
+  const allDisclosuresChecked = areDisclosuresChecked.every(Boolean);
+  const {isAdmin} = useAuth();
+
+  const {t} = useTranslation('administration');
+
+  /**
+   * Resets modal state values before closing the modal.
+   */
+  function handleClose(): void {
+    // Modal state
+    setAreDisclosuresChecked([false, false, false]);
+    onClose();
+  }
+
+  /**
+   * Marks data older than a defined time as ready for deletion by a Cloud Function
+   */
+  async function markOldDataForDeletion(): Promise<void> {
+    if (allDisclosuresChecked && isAdmin) {
+      const sensorsList = await firestore.collection('sensors').get();
+      const deletionDocRef = firestore.collection('deletion').doc('todo');
+
+      // Get current mapping or set to empty
+      const mapping =
+        (await (await deletionDocRef.get())?.data()?.data) ??
+        Object.create(null);
+
+      const offset = 7;
```

Comment:
nit: rename to something like `daysPrior`

Reply:
Also maybe this could be a parameter? Using a date selector like in Amber's?

Reply:
It could be a parameter, but I think the issue is we want to avoid them choosing anything too recent that it could mess with the website. Maybe we merge it like this but then that can be a bug we may or may not choose to fix before code freeze? Either way I will rename the variable

Reply:
Also, if the user can choose, then we run into the issue where if the sensor is already present with a certain date x to delete and then they put a new date y where y<x, which do we choose as the deletion date

Reply:
I like the idea of user input field (we can require that it is at least blank days ago). I agree it's def a bug fix, don't worry about it now, and I think that we go with the latest delete date if there's ever a disagreement.

Reply:
Submitting a bug sounds good to me. I think we could have a condition that you can only delete data that is over a week old, but a parameter could give liaisons the flexibility to only delete data that is 6 months old, a year old, etc. which I think would be good

Reply:
Can handle in later #552